### ISSUE TITLE
Mechanism for updating the status of video transcoding jobs

### ISSUE THREAD

**Issue by mbertrand** [mitodl/odl-video-service#58]
Currently, when a user visits a video detail page, if the video status is not yet 'Complete' or 'Error', a call will be made to AWS ElasticTranscoder to check on the status of the transcode job, and then update the video status based on the result.  This should be done automatically instead, and there are a few options for this:

1. Use SNS in conjunction with django-elastic-transcoder's 'endpoint' URL to send transcoding status updates directly to the app.  This should work fine on a live site, but would have to be manually simulated on a local development build.

2. Have SNS send status updates to an SQS message queue instead, and then have a recurring celery task to process the messages in the queue. 

3. Have a recurring celery task that queries the database for all videos whose status is not yet completed or failed, and then queries AWS ElasticTranscoder for the status of each one's transcode job for any possible updates.

4. The celery task that creates the transcode job could also continue running [while checking the job status until the job has completed or failed](http://boto3.readthedocs.io/en/latest/reference/services/elastictranscoder.html#waiters), and then update the video status based on that result.

**Comment by mbertrand** [comment]
@giocalitri  @pdpinch Testing an SNS HTTP endpoint locally (option 1) can be achieved by [using ngrok](https://aws.amazon.com/blogs/developer/testing-webhooks-locally-for-amazon-sns/).  This would require creating an SNS subscription to an ngrok URL that would change every time ngrok is started (unless we get a paid plan that enables more permanent URL's).  

```
Session Status                online
Account                       Matt Bertrand (Plan: Free)
Version                       2.2.8
Region                        United States (us)
Web Interface                 http://127.0.0.1:4040
Forwarding                    http://d2f9ac8c.ngrok.io -> localhost:8089
Forwarding                    https://d2f9ac8c.ngrok.io -> localhost:8089
```

**Comment by mbertrand** [comment]
Option 3 implemented in #90
